metricbeat/docs/fields.asciidoc

```diff
@@ -410,6 +411,90 @@ type: boolean
 If true this namespace is currently not allowing writes.
 
 
+--
+
+[[exported-fields-alias]]
+== Alias fields
+
+Alias fields for compatibility with 7.x.
+
+
+
+*`beat.name`*::
++
+--
+type: alias
+
+--
+
+*`beat.hostname`*::
```

Comment:
I'm a little confused by the semantics here.  Is `beat.hostname` the alias, or the field the alias will point to?  Reading the above:
```
   - name: host.hostname
      type: alias
      path: beat.hostname
```

makes me think that `host.hostname` is the alias that is going to point to the real field `beat.hostname`.  I know the ECS field name is `host.hostname`.  Is the plan in 6.x it'll be the alias, and in 7.x, it'll be the real field when the mapping support many : 1 (the 1 being the ECS field, the many being all variations that are used across all the beats).

Can you let me know the specific field that will require the `many : 1` mapping?  I thought I recalled Boaz saying something about filebeat, but looking at the fields (well at least the ones with `host` in them) I don't see where the issue specifically is. Just looks like all the beats have `beat.hostname` which would map to `host.hostname`, which should still be 1:1?  Even though `beat.hostname` is a field in many different indices... ?

Sorry for so many questions, just trying to wrap my head around all this as I think about what should be tested!

Reply:
The fields that are aliased here are for 6.x and are only 1:1 mappings to make them compatible with 7.x data for Logging UI and Infrastructure UI.

For the many:1 mappings which are only introduce in 7.x you can find some examples here: 

* https://github.com/elastic/beats/pull/9245
* https://github.com/elastic/beats/pull/9081

This form apache and nginx and you will see that these have many fields in common and the old fields are mapped to the same new fields.

Reply:
@Stacey-Gammon Here's my attempt at clarifying as well.

For 6.x and for 7.x, the aliases are flipping direction indeed.

6.x: new fields introduced as aliases to the old field they will eventually replace. This is for forward compatibility. We will start being able to depend on the new field names in the later releases in the 6.x line for those. This is only applicable for 1:1 mappings, however (explained below).

7.x: For all fields being renamed (1:1 or many to 1), the ECS field is now the canonical field holding the data, and old fields alias to the ECS field.

Many to 1 mappings: Filebeat modules are indeed where this is happening the most, but there may be other places. All modules used to nest their fields under a namespace `[module name].[dataset].[field]`. In many cases, different modules had perfectly equivalent fields: `apache2.access.url` and `nginx.access.url`. These two examples are becoming `url.original`. In 6.x we can't create a forward compatible alias to all of these values, as the alias can only point to one field. However in 7.x we'll have all of these old fields be aliases to the new canonical field. Nothing prevents having many alias fields pointing to the same field.